### ISSUE TITLE
Android: App exits fullscreen mode when splash is visible

### ISSUE THREAD

**Issue by distinctdan** [apache/cordova-plugin-splashscreen#180]
On android, when I have this plugin installed, my app shows the status and navigation bars while the splash screen is visible. When I uninstall this plugin, the bars are hidden while the app is starting. I'm using the `<preference name="Fullscreen" value="true" />` pref in my config.xml. I'm not using cordova-plugin-fullscreen.

**Comment by distinctdan** [comment]
I did some debugging; here's what's going on:
1) `CordovaActivity onCreate:` Cordova reads the "Fullscreen" preference. If true and you haven't also set the "FullscreenNotImmersive" preference, it sets `immersiveMode = true`. However, it never actually calls `setSystemUiVisibility()` to set `FLAG_FULLSCREEN`.
2) `SplashScreen showSplashScreen:` The splashscreen plugin is looking for `FLAG_FULLSCREEN`, but it hasn't been set yet.
3) `CordovaActivity onWindowFocusChanged:` The app has launched. Now cordova calls `setSystemUiVisibility()` with the fullscreen flags.

So I'm new to cordova's codebase, but looks like there are a couple of ways to fix it:
1) Make cordova set the window flags in `onCreate` in addition to setting them in `onWindowFocusChanged`. This seems like the best solution to me overall, but it requires changing cordova's code, which is outside this plugin.
2) Make this plugin read the fullscreen preference instead of relying on `FLAG_FULLSCREEN` being set.

Thoughts?

**Comment by Lirianna** [comment]
A work around for some Android phones, is to resize/make sure the splash is no larger than 1024x1024.
Since this is only working for some Android phones, I decided to remove splashscreen untill this is resolved.

**Comment by janpio** [comment]
> 1. `CordovaActivity onCreate:` Cordova reads the "Fullscreen" preference. If true and you haven't also set the "FullscreenNotImmersive" preference, it sets `immersiveMode = true`. However, it never actually calls `setSystemUiVisibility()` to set `FLAG_FULLSCREEN`

That could be considered a bug of `cordova-android`, correct? If so, it would be awesome if you could create a PR over there to fix this!

> Make this plugin read the fullscreen preference instead of relying on `FLAG_FULLSCREEN` being set.

Hmm, the fullscreen-preference is usually not part of that plugin, so this is a bit dirty. But if this fixes the behavior _right now_, I think having it read that **as well additionally** I think we should be fine. Can you create a PR for this as well?

> _EDIT_: Just tried both solutions, and settings the flags in onCreate helped, but the status bar and navigation bar still flash in/out when the app starts. It'd be great if someone with more experience in the codebase could take a look at it.

Not that many people of that around any more, so we have to work with what we got - and in this case you seem to be the expert for this problem @distinctdan - so I would really appreciate it if you could go ahead and suggest some code.

**Comment by distinctdan** [comment]
Alright, I'll consider looking into it more. I am planning on getting my splashscreen functionality worked out in the next few weeks or so, so I'll take a look and see if I can get it figured out.

**Comment by distinctdan** [comment]
Ok, got this one figured out. It's my first PR here, so I'm definitely open to feedback or adjustments. I'm going to do another PR over on `cordova-android` to fix the flags not getting set in `onCreate`.

**Comment by janpio** [comment]
Let's keep this open until the PR is actually merged.

Just to make sure: This can't be fixed in the splashscreen plugin itself somehow?

**Comment by distinctdan** [comment]
I haven't figured out a way to completely fix it in the splashscreen plugin. This PR keeps the bars hidden while the splash is visible, but when the cordova main activity is made visible, they briefly pop in. We could call `setSystemUiVisibility()` again after the splash is hidden, but that feels a little hacky to me, like it ought to be the main activity's responsibility to set its own ui visibility, rather than us doing it.

**Comment by janpio** [comment]
I have no knowledge of Android at all, but your argument makes sense to me.

**Comment by peitschie** [comment]
Hi! Is there something still blocking this from merging?
This makes the situation today much better, even with the momentary flash of the navigation controls when the splash screen hides...

Tested on Android 10, and this works no problems!

**Comment by peitschie** [comment]
Hi!

Just noting, I've tested this on Android 10, and it definitely improves the situation immensely for me.

Is this still being blocked by something? The code presented here makes good sense to me.
